client/components/data/purchases/edit-card-details/index.jsx

```diff
@@ -32,7 +32,7 @@ function getStateFromStores( props ) {
 
 function isDataLoading( state ) {
 	return (
-		! state.card ||
+		state.card.isFetching ||
```

Comment:
Sounds like this solves most of the issues reported in #1264 though? We should get this fix merged and open new issues for any remaining problems. Could you do that @stephanethomas?